6-create the request account page front end

kshah2020 opened the pull request:
## Changes
`App.tsx`
- replace Example Page with RequestAccountPage

`RequestAccountPage.tsx`
- main page containing front end to request a page
- component/features
  - fields for user to enter information 
  - radio buttons for create user as and priority fields
  - submits form/console logs when submit button is pressed
  - has popup appear to inform the request was sent
  - refreshes page to allow a new account to be set

`RequestAccountPage.css`
- css formatting for page/form fields to match figma layout

`popup.css`
- CSS styling for the popup (after submit is pressed), to appear as a popup window and match the styling/formatting on figma

## Dependencies
- n/a

andrewyang1015 committed:
1-handling-resources created a resource page and sample resource data

yumegan committed:
test change

yumegan committed:
display resource page. get resources from firestor

Commit:
creating file for page

Commit:
template for account page

andrewyang1015 committed:
6-request-account-page implemented the front-end page for account request and confirmation popup

Commit:
editing frontend, connecting popup, and form validation

yieb commented on jaba/src/pages/ResourcePage/ResourcePage.module.css:
Remove this file since this is a file not relevant for this specific PR

yieb commented on jaba/src/pages/ResourcePage/ResourcePage.tsx:
Remove this file since this is a file not relevant for this specific PR

yieb commented on jaba/src/pages/RequestAccount/RequestAccountPage.css:
For consistency, can you rename the file to end in module.css

yieb replied:
Also make sure to rebase with main to make sure the custom fonts load in properly

yieb commented on jaba/src/models/ResourceObject.ts:
Remove this file from this PR since this is a file not relevant for this specific PR

yieb commented on jaba/src/backend/firestoreFunctions.ts:
Remove this file from the PR since this the changes are not relevant for this specific PR

yieb commented on jaba/src/pages/RequestAccount/RequestAccountPage.css:
```diff
@@ -0,0 +1,123 @@
+body {
+    background-color: #012146;
```
change hex code to #023C47

yieb commented:
![image](https://github.com/Hack4Impact-UMD/jefferson-area-board-for-aging/assets/52669318/a9cc8951-9214-46ee-94d3-5795e8d125b2)

When I load the page, the sizing is off. Try to make design a bit more adaptable for different screen sizes

yieb commented:
![image](https://github.com/Hack4Impact-UMD/jefferson-area-board-for-aging/assets/52669318/7da111e6-2f5c-4368-b9a3-404d2df91a9b)

The size of the input boxes should not be able to be changed

yieb commented on jaba/src/pages/RequestAccount/RequestAccountPage.css:
```diff
@@ -0,0 +1,123 @@
+    width: 100px;
+    height: 40px;
+    flex-shrink: 0;
+    border-radius: 25px;
```
Change border-radius to 6px;

yieb commented:
![image](https://github.com/Hack4Impact-UMD/jefferson-area-board-for-aging/assets/52669318/ab07ad84-e2f0-4410-88e8-75b9800d8117)

The filler text inside of the input boxes is not centered like on the Figma design
The filler text in all the boxes should also match the filler text shown on the design.

yieb commented:
![image](https://github.com/Hack4Impact-UMD/jefferson-area-board-for-aging/assets/52669318/4e7c1f4a-f118-469b-b038-1ae446c8c99a)

![image](https://github.com/Hack4Impact-UMD/jefferson-area-board-for-aging/assets/52669318/8b11c56e-3dd3-41bf-b8ae-0c374808edb6)

- Layout of input boxes is different (Figma design has Position at Organization in second column)
- The margin between the two rows of inputs is different
- The Submit Request button should be centered with the input boxes in the second column
- The buttons for Priority and Create account as should be inline with the text

yieb commented on jaba/src/pages/RequestAccount/RequestAccountPage.css:
```diff
@@ -0,0 +1,123 @@
+}
+
+.request-header {
+    margin: 39px 769.57px 806px 62px;
```
Where are these numbers coming from?

yieb replied:
What is the different between request page and request header? They both appear to wrap all the HTML code

yieb commented on jaba/src/pages/RequestAccount/RequestAccountPage.tsx:
```diff
@@ -0,0 +1,217 @@
+              <p>
+                <span className="submitted">
+                  Submit another request?{' '}
+                  <a className="click-here" onClick={resetPage}>
```
Can we make the cursor change here to make it more obvious that you can click here?

yieb commented on jaba/src/pages/RequestAccount/RequestAccountPage.css:
```diff
@@ -0,0 +1,123 @@
+    line-height: normal;
+}
+
+.heading-container {
```
Where are all the px values in the file coming from? Are all of them pulled straight from Figma or are some just numbers that were arbitrarily set so it looked nice?

andrewyang03 replied:
They were generally pulled from the figma page

yieb commented:
Sidenote: adding a screenshot of what the frontend looks like on your end would be very helpful to include when making your PR next time

yieb reviewed (changes requested):
See comments

andrewyang1015 committed:
6-request-account-page implemented pr changes

andrewyang1015 committed:
6-request-account-page

andrewyang1015 committed:
6-request-account-page cleaned merge conflicts

andrewyang1015 committed:
6-request-account-page fixed fonts

andrewyang1015 committed:
6-request-account-page
Added font styles for popup as well

andrewyang1015 committed:
6-request-account-page
Centered text within input fields and had Click here in popup hover

andrewyang1015 committed:
6-request-account-page
Adjusted to style conventions

yieb committed:
Clean up CSS styling

- Remove redundant/dead CSS styling
- Update textarea tags in tsx file to input tags
- Fix an issue with ".priorityOptions" style being used for both a div
  and p tag

yieb committed:
Merge branch 'main' into 6-request-account-page

yieb commented:
Updated CSS styling. Final Result:
![image](https://github.com/Hack4Impact-UMD/jefferson-area-board-for-aging/assets/52669318/378413c2-f88f-48a4-a775-d02ab4417c94)
